### PR TITLE
Lockout users when Enterprise License expires, except Non-Admin and Pipeline users 

### PR DESCRIPTION
**Pull request by acohen4**
Desired Behavior:
When the Enterprise License expires all users are locked out (cannot authenticate), with the exception of 
- the Root User
- any user with the ClusterAdmin Role
- the PPS Master User
- any Pipeline

Implementation Line Items:
1.) Returning Enterprise Service to Side-car
2.) Adding a check for expired Cluster Admin.

### CODE REVIEW

**Comment by acohen4** [comment]
@actgardner the changes to this file may not be necessary. 

We are already checking for enterprise token validity during login in Authenticate(), with the method: `expiredEnterpriseCheck`

I added a change to test for Cluster Admin role instead of root user. This may not be right, but wanted to check with you.

I also put this check in getAuthenticatedUser(), this would lockout an already authenticated user. I don't know if that's overkill either, or whether there's some use-case where we would like to enable authenticated users to continue using the cluster with the expired license

**Comment by actgardner** [comment]
I think we should consolidate all this logic into one method:
- allowing all users with `clusterAdmin` makes sense
- we need to allow all PPS pipelines (not just the PPS superuser) so pipelines keep working

> I also put this check in getAuthenticatedUser(), this would lockout an already authenticated user. I don't know if that's overkill either, or whether there's some use-case where we would like to enable authenticated users to continue using the cluster with the expired license

That's basically the behaviour we want - if the user is already authenticated but the enterprise license has expired, we should block them from making API calls.

File: src/server/auth/server/api_server.go [hunk 1]
```diff
@@ -301,6 +305,58 @@ func (a *apiServer) getEnterpriseTokenState(ctx context.Context) (enterpriseclie
...
+		return errors.Wrapf(err, "error confirming Pachyderm Enterprise token")
+	}
+
+	if state == enterpriseclient.State_ACTIVE {
+		return nil
+	}
+
+	// this call internally checks for an Active License
+	// and Authorized Users in case of a deactivated license
+	if _, err = a.getAuthenticatedUser(ctx); err != nil {
```
[the remark this answers is not in the document]
Comment (actgardner) [reply src/server/auth/server/api_server.go:322]: > If I understand correctly, we have to first derive the username from OIDC or IdToken, and then check the associated roles. Is there a place where the RootUser, or PPSMaster is calling Authenticate with an auth token?

The users with a `pach:` prefix don't come from the IDP, so they never call `Authenticate`. This code path only applies to IDP users, who may be admins, but we don't know until they've authenticated. The old code is kind of misleading here, it was just a placeholder that preventing logging in when the license expired.

If we don't allow non-admin to _do_ anything, do we even need to special-case `Authenticate`? They could just log in and then whatever they try to do in pachctl will fail? That might be simplest solution.

**Review by actgardner — APPROVED** [review]
Awesome, this looks good!

**Pull request merged**